reana_server/scheduler.py

```diff
@@ -83,29 +103,44 @@ def doesnt_exceed_max_reana_workflow_count():
             .scalar()
         )
         if running_workflows >= REANA_MAX_CONCURRENT_BATCH_WORKFLOWS:
+            message = (
+                f"there are already {running_workflows} running workflows "
+                f"whilst the allowed maximum is {REANA_MAX_CONCURRENT_BATCH_WORKFLOWS}."
+            )
             doesnt_exceed = False
     except SQLAlchemyError as e:
-        logging.error(
-            "Something went wrong while querying for number of running workflows."
-        )
+        message = "Something went wrong while querying for number of running workflows."
+        logging.error(message)
         logging.error(e)
         doesnt_exceed = False
     Session.commit()
-    return doesnt_exceed
+    return doesnt_exceed, message
 
 
-def reana_ready(workflow_min_job_memory):
+def reana_ready(workflow_min_job_memory: Optional[float]) -> Tuple[bool, Optional[str]]:
     """Check if REANA can start new workflows."""
-    conditions = [check_predefined_conditions, doesnt_exceed_max_reana_workflow_count]
+    conditions_list = [
+        doesnt_exceed_max_reana_workflow_count,
+        partial(check_memory_availability, workflow_min_job_memory)
+        # check_predefined_conditions,
```

Comment:
Is this supposed to be commented? If we want to remove this check, I think it's better to just delete the whole function.

Reply:
I was hesitating to remove it in the beginning, since it was there for 3 years.. But indeed because of the reasons I explained in the MM I think it's better to get rid of this check